homeassistant/components/frontier_silicon/browse_media.py

```diff
@@ -0,0 +1,242 @@
+    )
+
+    response = BrowseMedia(
+        media_class=MEDIA_CLASS_DIRECTORY,
+        media_content_id=search_id,
+        media_content_type=search_type,
+        title=title,
+        can_play=bool(search_id),
+        can_expand=True,
+        children=children,
+        thumbnail=thumbnail,
+    )
+    response.calculate_children_class()
+
+    return response
+
+
+async def item_payload(item, get_thumbnail_url=None):
+    """
+    Create response payload for a single media item.
+
+    Used by async_browse_media.
+    """
+    title = item["label"]
+
+    media_class = None
+
+    if "key" in item:
+        media_content_type = f"{item['type']}"
+        media_content_id = f"{item['key']}"
+        can_play = True
+        can_expand = False
+    else:
+        # this case is for the top folder of each type
+        # possible content types: album, artist, movie, library_music, tvshow, channel
+        media_class = MEDIA_CLASS_DIRECTORY
+        media_content_type = item["type"]
+        media_content_id = ""
+        can_play = False
+        can_expand = True
+
+    if media_class is None:
+        try:
+            media_class = CHILD_TYPE_MEDIA_CLASS[media_content_type]
+        except KeyError as err:
+            _LOGGER.debug("Unknown media type received: %s", media_content_type)
+            raise UnknownMediaType from err
+
+    thumbnail = item.get("thumbnail")
+    if thumbnail is not None and get_thumbnail_url is not None:
+        thumbnail = await get_thumbnail_url(
+            media_content_type, media_content_id, thumbnail_url=thumbnail
+        )
+
+    return BrowseMedia(
+        title=title,
+        media_class=media_class,
+        media_content_type=media_content_type,
+        media_content_id=media_content_id,
+        can_play=can_play,
+        can_expand=can_expand,
+        thumbnail=thumbnail,
+    )
+
+
+async def library_payload():
+    """
+    Create response payload to describe contents of a specific library.
+
+    Used by async_browse_media.
+    """
+    library_info = BrowseMedia(
+        media_class=MEDIA_CLASS_DIRECTORY,
+        media_content_id="library",
+        media_content_type="library",
+        title="Media Library",
+        can_play=False,
+        can_expand=True,
+        children=[],
+    )
+
+    library = {
+        MEDIA_TYPE_CHANNEL: "Channels",
+        "preset": "Favorites",
+    }
+
+    library_info.children = await asyncio.gather(
+        *(
+            item_payload(
+                {
+                    "label": item["label"],
+                    "type": item["type"],
+                    "uri": item["type"],
+                },
+            )
+            for item in [
+                {"label": name, "type": type_} for type_, name in library.items()
+            ]
+        )
+    )
+
+    return library_info
+
+
+async def _get_content_list(
+    media_library, list_url, set_nav_state=True, max_results=500
+):
+    nav = None
+    if set_nav_state:
```

Comment:
This if-statement wraps the rest of the method so turn this into a guard clause.

```python
if not set_nav_state:
    return None, None
```